### PR TITLE
Add async_client

### PR DESCRIPTION
**Pull request by nikteliy**
- add async functions to async_client

- add tests

- remove async functions from client.py

- remove async tests from test.client.py

- add async tests to action

### CODE REVIEW

File: snap7/client_async.py [hunk 1]
```diff
@@ -3,36 +3,131 @@
...
+    def sync_to_async(self, func):
+        @wraps(func)
+        async def wrapper(*args, loop=None, executor=None, **kwargs):
+            if loop is None:
+                loop = asyncio.get_event_loop()
+            pfunc = partial(func, *args, **kwargs)
+            return await loop.run_in_executor(self.executor, pfunc)
+        return wrapper
 
+    async def as_compress(self, time) -> Awaitable[int]:
```
Comment (swamper123) [inline snap7/client_async.py:31]: I had a longer discussion once with @gijzelaerr about the topic where to store snap7-as_* methods.
The result was, that we left them in client.
I think it is possible to overload them in a Coroutine object without any trouble.

File: snap7/client_async.py [hunk 2]
```diff
@@ -3,36 +3,131 @@
...
-    async def f(*args, **kw):
-        code = await func(*args, **kw)
-        check_error(code, context="client")
+class ClientAsync(Client):
+    def __init__(self, max_workers=1):
+        super().__init__()
+        self.executor = ThreadPoolExecutor(max_workers=max_workers)
 
-    return f
+    def sync_to_async(self, func):
```
Comment (swamper123) [inline snap7/client_async.py:22]: That is a pretty smooth way to convert functions to coroutines tbh!
But in the end, aren't the Coroutines blocking other running Coroutines? Could be ProcessPoolExecutor be the better choice?

File: test/test_client_async.py [hunk 3]
```diff
@@ -1,28 +1,33 @@
...
+                        b'\x85V\x1f2\xa1\x00*\x00\x00\x00\x00\x00\x02\x01\x0f\x05c\x00#\x00\x00\x00' \
+                        b'\x11\x04\x10\x01\x04\x01\x04\x01\x04\x01\x04\x01\x04\x01\x04\x01\x04\x01' \
+                        b'\x04\x01\x04\x01\x04\x01\x04\x01\x04\x01\x04\x01\x04\x01\x04\x01\x04\x00' \
+                        b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' \
+                        b'\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 
-@pytest.fixture
-def testserver():
+
+@pytest.fixture(scope="session")
```
[the remark this answers is not in the document]
Comment (nikteliy) [reply test/test_client_async.py:29]: Good point. I'll fix it.

File: snap7/client_async.py (hunk 1 above)
Comment (nikteliy) [reply snap7/client_async.py:31]: I'll do as you say guys. Both options are good for me.

File: snap7/client_async.py (hunk 2 above)
Comment (nikteliy) [reply snap7/client_async.py:22]: As far as I know, I/O calls release GIL, so a thread should be enough. But maybe you're right. I'll look into it.

Comment (swamper123) [reply snap7/client_async.py:22]: I made a research and it seems that you are right about that. I/O bounded calls seem to be the exception in the whole Thread case,.

**Pull request closed without merging**